lib/stats/DefaultStatsFactoryPlugin.js

```diff
@@ -862,16 +888,22 @@ const SIMPLE_EXTRACTORS = {
 			if (errorDetails === "auto") {
 				if (warnings) {
 					const warnings = cachedGetWarnings(compilation);
-					object.filteredWarningDetailsCount = warnings
-						.map(e => typeof e !== "string" && e.details)
-						.filter(Boolean).length;
+					const filtered = object.filteredWarningDetailsCount || 0;
```

Comment:
It doesn't seem right to read from `object`.

Best move the whole `errorsDetails` logic here into `errors`/`warnings` and write `filteredWarningDetailsCount` only there.